src/reconcile/rec_visibility.c

```diff
@@ -182,6 +182,134 @@ __rec_append_orig_value(
     return (ret);
 }
 
+/*
+ * __rec_delete_hs_record --
+ *     Delete the update left in the history store
+ */
+static int
+__rec_delete_hs_record(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ITEM *key,
+  WT_UPDATE *delete_upd, WT_UPDATE *delete_tombstone)
+{
+    WT_DECL_RET;
+    bool hs_read_committed;
+#ifdef HAVE_DIAGNOSTIC
+    WT_TIME_WINDOW *hs_tw;
+#endif
+
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+    hs_read_committed = F_ISSET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+
+    /* No need to delete from the history store if it is already obsolete. */
+    if (delete_tombstone != NULL && __wt_txn_upd_visible_all(session, delete_tombstone)) {
+        ret = 0;
```

Comment:
I guess it is to make it more explicit, but don't we have the guarantee that `ret` is `0` here?

Reply:
removed